src/PanoImageRenderer/PanoImageRenderer.js

```diff
@@ -288,6 +309,50 @@ export default class PanoImageRenderer extends Component {
 		return true;
 	}
 
+	initShaderProgram() {
+		const gl = this.context;
+
+		if (this.shaderProgram) {
+			gl.deleteProgram(this.shaderProgram);
+			this.shaderProgram = null;
+		}
+
+		const renderer = this._renderer;
+
+		const vsSource = renderer.getVertexShaderSource();
+		const fsSource = renderer.getFragmentShaderSource();
+
+		const vertexShader = WebGLUtils.createShader(gl, gl.VERTEX_SHADER, vsSource);
+		const fragmentShader = WebGLUtils.createShader(gl, gl.FRAGMENT_SHADER, fsSource);
+
+		const shaderProgram = WebGLUtils.createProgram(gl, vertexShader, fragmentShader);
+
+		if (!shaderProgram) {
+			throw new Error(`Failed to intialize shaders: ${WebGLUtils.getErrorNameFromWebGLErrorCode(gl.getError())}`);
+		}
+
+		gl.useProgram(shaderProgram);
+		shaderProgram.vertexPositionAttribute = gl.getAttribLocation(shaderProgram, "aVertexPosition");
+		gl.enableVertexAttribArray(shaderProgram.vertexPositionAttribute);
+		shaderProgram.pMatrixUniform = gl.getUniformLocation(shaderProgram, "uPMatrix");
+		shaderProgram.mvMatrixUniform = gl.getUniformLocation(shaderProgram, "uMVMatrix");
+		shaderProgram.samplerUniform = gl.getUniformLocation(shaderProgram, "uSampler");
+		shaderProgram.textureCoordAttribute = gl.getAttribLocation(shaderProgram, "aTextureCoord");
+
+		gl.enableVertexAttribArray(shaderProgram.textureCoordAttribute);
+
+		// clear buffer
+		gl.clear(gl.COLOR_BUFFER_BIT | gl.DEPTH_BUFFER_BIT | gl.STENCIL_BUFFER_BIT);
+		// Use TEXTURE0
+		gl.uniform1i(shaderProgram.samplerUniform, 0);
+
+		if (!shaderProgram) {
```

Comment:
@WoodNeck 
This is a duplicate code.

Reply:
Looks like the latter one is added by a mistake, good check.